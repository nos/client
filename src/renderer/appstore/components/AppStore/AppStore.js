import React from 'react';
import { arrayOf, object, string } from 'prop-types';

import TopBackground from 'shared/images/appstore/bg-shape-1.svg';

import CategoryTitle from '../CategoryTitle';
import AppButton from '../AppButton';

import styles from './AppStore.scss';

const AppStore = ({ apps, categories }) => (
  <div className={styles.appstore}>
    <div className={styles.top}><TopBackground /></div>
    {categories.map((cat) => (
      <React.Fragment key={cat}>
        <CategoryTitle key={cat}>{cat}</CategoryTitle>
        <div className={styles.grid}>
          {apps
            .filter((app) => app.category === cat)
            .map((app) => <AppButton key={app.title} app={app} />)
          }
        </div>
      </React.Fragment>
    ))}
  </div>
);

AppStore.propTypes = {
  apps: arrayOf(object),
  categories: arrayOf(string)
};

AppStore.defaultProps = {
  apps: [],
  categories: []
};

export default AppStore;
