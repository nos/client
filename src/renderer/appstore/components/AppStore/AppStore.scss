.appstore {
  -webkit-font-smoothing: antialiased;
  padding: 24px;

  .privateTabLink {
    padding: 0px;
  }

  .grid {
    display: grid;
    grid-gap: 16px;
    grid-template-columns: repeat(auto-fill, 380px);
    margin-bottom: 32px;
  }

  .top {
    position: sticky;
    height: 0;
    top: 0;
    transform: translate(-24px, -24px);

    svg {
      z-index: -1;
      max-width: 90%;
      height: auto;
    }
  }
}

@media (max-width: 900px) {
  .appstore {
    .grid {
      grid-template-columns: repeat(auto-fill, 190px);
    }
  }
}
