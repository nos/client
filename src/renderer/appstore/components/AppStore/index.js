import { compose } from 'recompose';
import { withData, withProgressComponents, LOADING, FAILED } from 'spunky';

import withInitialCall from 'shared/hocs/withInitialCall';
import { withErrorToast } from 'shared/hocs/withToast';
import Loading from 'shared/components/Loading';
import withProgressChange from 'shared/hocs/withProgressChange';

import AppStore from './AppStore';
import getAppStoreData from '../../actions/getAppStoreData';

export default compose(
  withInitialCall(getAppStoreData),
  withProgressComponents(getAppStoreData, {
    [LOADING]: Loading
  }),
  withErrorToast(),
  withProgressChange(getAppStoreData, FAILED, (state, props) => {
    props.showErrorToast(
      `Loading is taking longer than expected. Please try again later. ${state.error}`
    );
  }),
  withData(getAppStoreData)
)(AppStore);
