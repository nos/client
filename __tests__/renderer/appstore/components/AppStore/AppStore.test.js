import React from 'react';
import { mount } from 'enzyme';

import AppStore from 'appstore/components/AppStore/AppStore';

describe('<AppStore />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<AppStore />);
    expect(wrapper).toMatchSnapshot();
  });
});
