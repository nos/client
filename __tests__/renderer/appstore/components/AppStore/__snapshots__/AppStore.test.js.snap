// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppStore /> renders correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "apps": Array [],
      "categories": Array [],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="appstore"
      >
        <div
          class="top"
        >
          <svg />
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="top"
          >
            <Unknown />
          </div>,
          Array [],
        ],
        "className": "appstore",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="top"
          >
            <svg />
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown />,
            "className": "top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": Object {
                "instance": <svg />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "svg",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "apps": Array [],
        "categories": Array [],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="appstore"
        >
          <div
            class="top"
          >
            <svg />
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="top"
            >
              <Unknown />
            </div>,
            Array [],
          ],
          "className": "appstore",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="top"
            >
              <svg />
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Unknown />,
              "className": "top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": Object {
                  "instance": <svg />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "svg",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          null,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__linkedRoots__): Array [],
  Symbol(enzyme.__unrendered__): <AppStore
    apps={Array []}
    categories={Array []}
  />,
  Symbol(enzyme.__updatedBy__): null,
}
`;
